src/libraries/System.Data.OleDb/src/OleDbDataReader.cs

```diff
@@ -2257,21 +2262,22 @@ private void AppendSchemaUniqueIndexAsKey(Hashtable baseColumnNames, object[] re
             }
         }
 
-        private MetaData FindMetaData(string name)
+        private MetaData? FindMetaData(string name)
         {
-            int index = _fieldNameLookup.IndexOfName(name);
-            return ((-1 != index) ? _metadata[index] : null);
+            int index = _fieldNameLookup!.IndexOfName(name);
+            return ((-1 != index) ? _metadata![index] : null);
         }
 
-        internal void DumpToSchemaTable(UnsafeNativeMethods.IRowset rowset)
+        internal void DumpToSchemaTable(UnsafeNativeMethods.IRowset? rowset)
         {
             List<MetaData> metainfo = new List<MetaData>();
 
-            object hiddenColumns = null;
+            object? hiddenColumns = null;
             using (OleDbDataReader dataReader = new OleDbDataReader(_connection, _command, int.MinValue, 0))
             {
                 dataReader.InitializeIRowset(rowset, ChapterHandle.DB_NULL_HCHAPTER, IntPtr.Zero);
-                dataReader.BuildSchemaTableInfo(rowset, true, false);
+                // TODO: BuildSchemaTableInfo asserts that rowset isn't null, but doesn't do anything with it
```

Comment:
Nit: another `TODO` that could be `TODO-NULLABLE` if you want.

```suggestion
                // TODO-NULLABLE: BuildSchemaTableInfo asserts that rowset isn't null, but doesn't do anything with it
```